### PR TITLE
Move validation problems into result response

### PR DESCRIPTION
**Pull request by frederike-ramin**
# Short Description
- Before, the validation problems were stored in the `error_message` column of the result table. However, as this had a VARCHAR datatype they were stored like this: `"{"Error 1", "Error 2"}`
- Now, they instead get stored in the `result` column

# Changes
- Store validation_problems as dict in result + return as object
- Adapt test for non-validation errors no not include validation_problems

# Feedback
- Do we have any end-to-end tests that I should adapt to be sure this actually works as intended? Because so far, I think e.g. the ResponseDto is never actually tested?

# How to review this Pull Request
[Link to Confluence](https://digitalservice4germany.atlassian.net/wiki/spaces/STL/pages/117637157/Development+Process#Approve-or-comment) (internal)

### CODE REVIEW

File: erica/application/JobService/job.py [hunk 1]
```diff
@@ -45,7 +45,9 @@ async def perform_job(request_id: UUID, repository: base_repository_interface, s
                 exc_info=True
             )
             entity.error_code = e.generate_error_response().get('message')
-            entity.error_message = e.generate_error_response().get('validation_problems', e.generate_error_response().get('message'))
+            entity.error_message = e.generate_error_response().get('message')
```
Comment (VictorDelCampo) [inline erica/application/JobService/job.py:48]: I know this was already there before, but why do we call `e.generate_error_response()` several times? Couldn't we just do something like `error_response = e.generate_error_response()` and then extract message and validation problems from there?

**Review by VictorDelCampo — APPROVED** [review]
LGMT, just 1 small comment! 

What about the FSC actions? Can they get any validation errors? They can't right?

And surely we would need a new integration test where we maybe send a tax declaration with a validation error (like account holder person_b without defining person_b data?) and then test that the state is FAILED and the validations problems are in the result attribute?

**Review by JulianRoesner — APPROVED** [review]
LGTM 🚀.

> Do we have any end-to-end tests that I should adapt to be sure this actually works as intended? Because so far, I think e.g. the ResponseDto is never actually tested?

Yes, you are right. Ideally, we would have some kind of integration tests (similar to the ones for the job) for the api. These should test how api and database work together (does the api return the correct result for some kind of data in the database). We could also add a contract test for grundsteuer and est and check the validation errors, wdyt?

File: erica/application/JobService/job.py (hunk 1 above)
Comment (frederike-ramin) [reply erica/application/JobService/job.py]: Good remark. 👍

**Comment by frederike-ramin** [comment]
> And surely we would need a new integration test where we maybe send a tax declaration with a validation error (like account holder person_b without defining person_b data?) and then test that the state is FAILED and the validations problems are in the result attribute?

Added precisely that test. Works like a charm (after merging the retry PR #103).

File: contract_tests/test_integration_queue.py [hunk 2]
```diff
@@ -728,6 +794,20 @@ def test_if_get_existing_request_then_return_200_and_response_with_correct_param
...
+        uuid = response.headers['Location'].split("/")[2]
+        sleep(6)
+        response = requests.get(ERICA_TESTING_URL + self.endpoint + "/" + uuid)
+        assert response.status_code == 200
+        assert response.json()["processStatus"] == "Failure"
+        assert response.json()["result"] is not None
+        assert response.json()["result"]["validation_errors"] is not None
+        assert response.json()["errorCode"] is not None
+        assert response.json()["errorMessage"] is not None
+
```
Comment (JulianRoesner) [inline contract_tests/test_integration_queue.py:810]: Could we maybe have a similar test for est? I can also add that if you want me to

Comment (frederike-ramin) [reply contract_tests/test_integration_queue.py:810]: I already added it one commit prior:https://github.com/digitalservice4germany/erica/pull/110/commits/7cae9d0cda6f042d82ef1850017c56dd386b1a70

**Pull request merged**